driver/razermouse_driver.c

```diff
@@ -3607,9 +3619,11 @@ static int razer_mouse_probe(struct hid_device *hdev, const struct hid_device_id
             CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_device_idle_time);
             CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_dpi);
             CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_logo_led_brightness);
-            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_logo_matrix_effect_breath);
-            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_logo_matrix_effect_static);
-            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_logo_matrix_effect_none);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_scroll_matrix_effect_spectrum);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_scroll_matrix_effect_reactive);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_scroll_matrix_effect_breath);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_scroll_matrix_effect_static);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_scroll_matrix_effect_none);
             break;
```

Comment:
Are you modifying existing devices on purpose? I've seen the crossed out note in the top-most message, please clarify.

Reply:
According to Razer's documentation all three versions of the Viper should support the same effects. I initially separated the Viper Mini from the other two models as that was the only device I have to test with. It supports spectrum and reactive through my testing. Then in #1164 @troudbal confirmed the Viper also supports the same effects, so I went ahead and included them together.

The Ultimate has admittedly not been tested. I'm fairly certain the Ultimate supports spectrum and reactive just the same as the other two. I could separate the Ultimate out from the Viper and Viper Mini if you want.

Reply:
fine then